### PR TITLE
Soc 1690

### PR DESCRIPTION
**Pull request by qaga**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.wikia.webdriver.pageobjectsfactory.pageobject.discussions;
+
+import com.wikia.webdriver.common.core.configuration.Configuration;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.configuration.Configuration'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import com.wikia.webdriver.common.core.url.UrlBuilder;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.url.UrlBuilder'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    driver.get(urlBuilder.getUrlForWiki().replace("/wiki", "") + String.format(PATH, wikiID));
+    return this;
+  }
+
+  public PostsListPage open() {
+    return open(DEFAULT_ID);
+  }
+
+  public boolean isPostListNotEmpty() {
+    return postList.size() > 0;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1155)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/332439601b0981f6f721554fc8ca7620cf98c9a2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/332439601b0981f6f721554fc8ca7620cf98c9a2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostsListPage.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+/**
+ * Created by qaga on 2015-11-25.
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java]: Use `@ownership Social Wikia` instead

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  public PostsListPage open() {
+    return open(DEFAULT_ID);
+  }
+
+  public boolean isPostListNotEmpty() {
+    return postList.isEmpty();
+  }
+
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java]: remove this extra line

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostDetailsPage.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package com.wikia.webdriver.pageobjectsfactory.pageobject.mercury.discussions;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import org.openqa.selenium.WebDriver;
+
+/**
+ * Created by qaga on 2015-12-01.
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostDetailsPage.java]: Use `@ownership Social Wikia` instead

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostDetailsPage.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  public PostDetailsPage(WebDriver driver) {
+    super(driver);
+  }
+
+  public PostDetailsPage openPost(String wikiID) {
+    driver.get(urlBuilder.getUrlForWiki().replace("/wiki", "") + String.format(PATH, wikiID));
+    return this;
+  }
+
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostDetailsPage.java]: remove extra line

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostsListPage.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+/**
+ * Created by qaga on 2015-12-01.
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostsListPage.java]: Use `@ownership Social Wikia` instead

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostsListPage.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+ */
+public class PostsListPage extends WikiBasePageObject {
+  @FindBy(css = ".post-detail")
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostsListPage.java]: please provide extra line above this

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostsListPage.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+  public PostsListPage open() {
+    return open(DEFAULT_ID);
+  }
+
+  public boolean isPostListNotEmpty() {
+    return postList.isEmpty();
+  }
+
+
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostsListPage.java]: remove 2 extra lines

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostDetailAnon.java [hunk 11]
```diff
@@ -0,0 +1,8 @@
+package com.wikia.webdriver.testcases.discussions.mobile;
+
+/**
+ * Created by qaga on 2015-11-24.
+ */
+public class PostDetailAnon {
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostDetailAnon.java]: Why you need empty class? Best practice is to not create placeholders like this. Please remove this and create when you will need it.

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostDetailLoggedUser.java [hunk 12]
```diff
@@ -0,0 +1,8 @@
+package com.wikia.webdriver.testcases.discussions.mobile;
+
+/**
+ * Created by qaga on 2015-11-24.
+ */
+public class PostDetailLoggedUser {
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostDetailLoggedUser.java]: Why you need empty class? Best practice is to not create placeholders like this. Please remove this and create when you will need it.

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+public class PostsListAnon extends NewTestTemplate {
+
+  @Test
+  @Execute(asUser = User.ANONYMOUS)
+  public void postsListLoads() {
+    PostsListPage postsList = new PostsListPage(driver).open();
+    postsList.isPostListNotEmpty();
+  }
+
+
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java]: remove extra 2 lines

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+import com.wikia.webdriver.common.core.annotations.User;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.mercury.discussions.PostsListPage;
+
+import org.testng.annotations.Test;
+
+/**
+ * Created by qaga on 2015-11-24.
+ */
+@Test(groups = {"Mercury", "Discussions"})
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java]: Don't add `Mercury` group, we will drop that group soon (for now it is not used)
Just use `@Test(groups = "Discussions")`

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.annotations.User;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.mercury.discussions.PostsListPage;
+
+import org.testng.annotations.Test;
+
+/**
+ * Created by qaga on 2015-11-24.
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java]: Use `@ownership Social Wikia` instead

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+ */
+@Test(groups = {"Mercury", "Discussions"})
+public class PostsListAnon extends NewTestTemplate {
+
+  @Test
+  @Execute(asUser = User.ANONYMOUS)
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java]: you don't need to put `User.ANONYMOUS` by default tests starts as anonymous

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListLoggedUser.java [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.annotations.User;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.mercury.discussions.PostsListPage;
+
+import org.testng.annotations.Test;
+
+/**
+ * Created by qaga on 2015-11-24.
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListLoggedUser.java]: Use `@ownership Social Wikia` instead

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListLoggedUser.java [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+ */
+public class PostsListLoggedUser extends NewTestTemplate{
+
+  @Test
+  @Execute(asUser = User.USER_3)
+  public void postsListLoads() {
+    PostsListPage postsList = new PostsListPage(driver).open();
+    postsList.isPostListNotEmpty();
+  }
+
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListLoggedUser.java]: remove extra line

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListLoggedUser.java [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Created by qaga on 2015-11-24.
+ */
+public class PostsListLoggedUser extends NewTestTemplate{
+
+  @Test
+  @Execute(asUser = User.USER_3)
+  public void postsListLoads() {
+    PostsListPage postsList = new PostsListPage(driver).open();
+    postsList.isPostListNotEmpty();
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListLoggedUser.java]: Wrap this in Assertion

File: src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+  public void postsListLoads() {
+    PostsListPage postsList = new PostsListPage(driver).open();
+    postsList.isPostListNotEmpty();
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/discussions/mobile/PostsListAnon.java]: Wrap this in Assertion

**Comment by tomasznapieralski** [comment]
Add testcases class to `TestSuite.xml`

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+}
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java]: you removed too many lines haha

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostDetailsPage.java [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+}
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostDetailsPage.java]: after `}` should be new line

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostsListPage.java [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  public PostsListPage open() {
+    return open(DEFAULT_ID);
+  }
+
+  public boolean isPostListEmpty() {
+    return postList.isEmpty();
+  }
+
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/discussions/PostsListPage.java]: remove this line

**Comment by tomasznapieralski** [comment]
LGTM :+1:

**Pull request merged**